File: Property_management_system/src/com/property/management/Payment.java
package com.property.management;

import java.util.ArrayList;
import java.util.List;

public class Payment {
    private String paymentID;
    private String tenantID;
    private String leaseID;
    private double amount;
    private String date;
    public static List<Payment> payments = new ArrayList<>();

    public Payment(String paymentID, String tenantID, String leaseID, double amount, String date) {
        this.paymentID = paymentID;
        this.tenantID = tenantID;
        this.leaseID = leaseID;
        this.amount = amount;
        this.date = date;
    }

    // Getters and Setters
    public String getPaymentID() {
        return paymentID;
    }

    public void setPaymentID(String paymentID) {
        this.paymentID = paymentID;
    }

    public String getTenantID() {
        return tenantID;
    }

    public void setTenantID(String tenantID) {
        this.tenantID = tenantID;
    }

    public String getLeaseID() {
        return leaseID;
    }

    public void setLeaseID(String leaseID) {
        this.leaseID = leaseID;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    // Methods
    public static void makePayment(Payment payment) {
        payments.add(payment);
        System.out.println("Payment made: " + payment.paymentID);
    }

    public static void viewPaymentHistory(String tenantID) {
        for (Payment payment : payments) {
            if (payment.tenantID.equals(tenantID)) {
                System.out.println("Payment: " + payment.amount + " on " + payment.date);
            }
        }
    }
}

File: src/payment.py
class Payment:
    payments = []

    def __init__(self, payment_id, tenant_id, lease_id, amount, date):
        self.payment_id = payment_id
        self.tenant_id = tenant_id
        self.lease_id = lease_id
        self.amount = amount
        self.date = date

    @classmethod
    def make_payment(cls, payment):
        cls.payments.append(payment)
        print(f"Payment {payment.payment_id} made.")

    @classmethod
    def view_payment_history(cls, tenant_id):
        for payment in cls.payments:
            if payment.tenant_id == tenant_id:
                print(f"Payment: ${payment.amount} on {payment.date}")
